shanoir-ng-front/src/app/shared/mass-download/mass-download.service.ts

```diff
@@ -65,148 +67,69 @@ export class MassDownloadService {
-            if (ret != 'cancel') {
-                return datasetsPromise.then(datasets => {
-                    this._downloadDatasets(datasets, ret.format, ret.nbQueues, ret.unzip, downloadState);
-                })
-            } else return Promise.resolve();
-        }).catch(error => {
-            if (error == this.BROWSER_COMPAT_ERROR_MSG) {
-                return datasetsPromise.then(datasets => {
-                    return this.openAltModal(format, options).then(ret => {
-                        if (ret != 'cancel') {
-                            return this._downloadAlt('datasetIds', datasets.map(ds => ds.id), ret, downloadState);
-                        } else return Promise.resolve();
-                    });
-                });
-            } else throw error;
-        });
+    downloadAllByAcquisitionId(acquisitionId: number, format?: Format, downloadState?: TaskState) {
+        return this.datasetService.getByAcquisitionId(acquisitionId).then(datasets => this.downloadByDatasets(datasets, format, downloadState));
     }
 
-    // zip only
-    downloadAllByAcquisitionId(acquisitionId: number, options?: DownloadSetupOptions, downloadState?: TaskState) {
-        return this.openAltModal(null, options, false).then(ret => {
-            if (ret != 'cancel') {
-                this.datasetService.downloadDatasetsByAcquisition(acquisitionId, ret, downloadState);
-            } else return Promise.resolve();
-        });
+    downloadAllByStudyIdAndSubjectId(studyId: number, subjectId: number, format?: Format, downloadState?: TaskState): Promise<void> {
+        return this.datasetService.getByStudyIdAndSubjectId(studyId, subjectId).then(datasets => this.downloadByDatasets(datasets, format, downloadState));
     }
 
-    // zip only
-    downloadDataset(datasetId: number, options: DownloadSetupOptions, downloadState?: TaskState) {
-        return this.openAltModal(null, options, false).then(ret => {
-            if (ret != 'cancel') {
-                this.datasetService.downloadFromId(datasetId, ret, null, downloadState);
-            } else return Promise.resolve();
-        });
+    downloadByIds(datasetIds: number[], format?: Format, downloadState?: TaskState): Promise<void> {
+        return this.datasetService.getByIds(new Set(datasetIds)).then(datasets => this.downloadByDatasets(datasets, format, downloadState));
     }
 
-    downloadAllByStudyIdAndSubjectId(studyId: number, subjectId: number): Promise<void> {
-        return this.openModal().then(ret => {
+    /**
+     * This method is the generic entry to download multiple datasets.
+     */
+    private downloadByDatasets(datasets: Dataset[], format?: Format, downloadState?: TaskState): Promise<void> {
+
+        // Here check if we have DICOM datasets or not
+        let hasDicoms: boolean = this.hasDicom(datasets);
+        if (!hasDicoms || format == "dcm") {
+
+            // Directly download if there are no dicom
```

Comment:
Yep, we can discuss this. I took this decision to simplify the download of such datasets, otherwise we would open the window just to choose to unzip or not. But this can be interesting.
The other solution would be to display for example the total size of download, so that we have a relevant information in this window